lib/models/application.coffee

```diff
@@ -130,6 +130,51 @@ getApplicationModel = (deps, opts) ->
 
 		.asCallback(callback)
 
+	###*
+	# @summary Get applications and their devices, along with each device's
+	# associated services' essential details
+	# @name getAllWithDeviceServiceDetails
+	# @public
+	# @function
+	# @memberof balena.models.application
+	#
+	# @description
+	# This method does not map exactly to the underlying model: it runs a
+	# larger prebuilt query, and reformats it into an easy to use and
+	# understand format. If you want more control, or to see the raw model
+	# directly, use `application.getAll(options)` instead.
+	#
+	# @param {Object} [options={}] - extra pine options to use
+	# @fulfil {Object[]} - applications
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.application.getAllWithDeviceServiceDetails().then(function(applications) {
+	# 	console.log(applications);
+	# })
+	#
+	# @example
+	# balena.models.application.getAllWithDeviceServiceDetails(function(error, applications) {
+	# 	if (error) throw error;
+	# 	console.log(applications);
+	# });
+	###
+	exports.getAllWithDeviceServiceDetails = (options = {}, callback) ->
+		callback = findCallback(arguments)
+
+		serviceOptions = mergePineOptions
+			$expand: [
+				owns__device: getCurrentServiceDetailsPineOptions()
+			]
+		, options
+
+		exports.getAll(serviceOptions)
+		.then (apps) ->
+			return apps.map (app) ->
```

Comment:
NIT: we probably don't need to use `.map` here. Iterating with `forEach` to set the `owns__device` property on the existing objects and returning the `apps` should be fine and would save an unnecessary array creation.